src/analysis/analytic/composition.lean

```diff
@@ -135,43 +157,77 @@ begin
   map_add'  := λ v i x y, by simp only [apply_composition_update,
     continuous_multilinear_map.map_add],
   map_smul' := λ v i c x, by simp only [apply_composition_update,
     continuous_multilinear_map.map_smul] }
 
-/-- The norm of `q.comp_along_composition_multilinear p c` is controlled by the product of
-the norms of the relevant bits of `q` and `p`. -/
+/-- The norm of `f.comp_along_composition_multilinear p c` is controlled by the product of
+the norms of the relevant bits of `f` and `p`. -/
 lemma comp_along_composition_multilinear_bound {n : ℕ}
-  (q : formal_multilinear_series 𝕜 F G) (p : formal_multilinear_series 𝕜 E F)
-  (c : composition n) (v : fin n → E) :
-  ∥q.comp_along_composition_multilinear p c v∥ ≤
-    ∥q c.length∥ * (∏ i, ∥p (c.blocks_fun i)∥) * (∏ i : fin n, ∥v i∥) :=
-calc ∥q.comp_along_composition_multilinear p c v∥ = ∥q c.length (p.apply_composition c v)∥ : rfl
-... ≤ ∥q c.length∥ * ∏ i, ∥p.apply_composition c v i∥ : continuous_multilinear_map.le_op_norm _ _
-... ≤ ∥q c.length∥ * ∏ i, ∥p (c.blocks_fun i)∥ *
+  (p : formal_multilinear_series 𝕜 E F) (c : composition n)
+  (f : continuous_multilinear_map 𝕜 (λ (i : fin c.length), F) G) (v : fin n → E) :
+  ∥f.comp_along_composition_multilinear p c v∥ ≤
+    ∥f∥ * (∏ i, ∥p (c.blocks_fun i)∥) * (∏ i : fin n, ∥v i∥) :=
+calc ∥f.comp_along_composition_multilinear p c v∥ = ∥f (p.apply_composition c v)∥ : rfl
+... ≤ ∥f∥ * ∏ i, ∥p.apply_composition c v i∥ : continuous_multilinear_map.le_op_norm _ _
+... ≤ ∥f∥ * ∏ i, ∥p (c.blocks_fun i)∥ *
         ∏ j : fin (c.blocks_fun i), ∥(v ∘ (c.embedding i)) j∥ :
   begin
     apply mul_le_mul_of_nonneg_left _ (norm_nonneg _),
     refine finset.prod_le_prod (λ i hi, norm_nonneg _) (λ i hi, _),
     apply continuous_multilinear_map.le_op_norm,
   end
-... = ∥q c.length∥ * (∏ i, ∥p (c.blocks_fun i)∥) *
+... = ∥f∥ * (∏ i, ∥p (c.blocks_fun i)∥) *
         ∏ i (j : fin (c.blocks_fun i)), ∥(v ∘ (c.embedding i)) j∥ :
   by rw [finset.prod_mul_distrib, mul_assoc]
-... = ∥q c.length∥ * (∏ i, ∥p (c.blocks_fun i)∥) * (∏ i : fin n, ∥v i∥) :
+... = ∥f∥ * (∏ i, ∥p (c.blocks_fun i)∥) * (∏ i : fin n, ∥v i∥) :
   by { rw [← c.blocks_fin_equiv.prod_comp, ← finset.univ_sigma_univ, finset.prod_sigma],
        congr }
 
+/-- Given a formal multilinear series `p`, a composition `c` of `n` and a continuous multilinear
+map `f` in `c.length` variables, one may form a continuous multilinear map in `n` variables by
+applying the right coefficient of `p` to each block of the composition, and then applying `f` to
+the resulting vector. It is called `f.comp_along_composition p c`. It is constructed from the
+analogous multilinear function `f.comp_along_composition_multilinear p c`, together with a norm
+control to get the continuity. -/
+def comp_along_composition {n : ℕ}
+  (p : formal_multilinear_series 𝕜 E F) (c : composition n)
+  (f : continuous_multilinear_map 𝕜 (λ (i : fin c.length), F) G) :
+  continuous_multilinear_map 𝕜 (λ i : fin n, E) G :=
```

Comment:
It's just that `comp_along_composition_multilinear` is just an intermediate step to be able to construct `comp_along_composition`, so that's why I gave it the longer name. I could also have called it `called_along_composition_aux`, or whatever. Once `comp_along_composition` is constructed, I never need to use  `comp_along_composition_multilinear` again. If you have an idea of a better name, I can change it (but I'd like to keep the shorter name for the most used version).

Reply:
I see!  I think `comp_along_composition_aux` would make the auxiliary nature clearer, but I don't really care.

Reply:
Renamed.